### PR TITLE
docker-machine 0.1.0-rc2

### PR DESCRIPTION
**Pull request by DanLipsitt**
Bump docker-machine to latest release candidate.

Missing a couple of go_resource sha1 sums. Is there a tool to help with that?

### CODE REVIEW

**Comment by DomT4** [comment]
That's a solid point. I've never seen Go resources checksummed in Homebrew/Homebrew before, @radeksimko how did you generate the checksums for those resources? They work, so I'm presuming they are correct, but I'm not quite sure how, to be honest. It's not the SHA of the commit reference, obviously, but I'm not quite sure what else it could be.

File: Formula/docker-machine.rb [hunk 1]
```diff
@@ -89,6 +89,26 @@ class DockerMachine < Formula
...
+  end
+
+  go_resource "github.com/rackspace/gophercloud" do
+    url "https://github.com/rackspace/gophercloud.git",
+        :revision => "2e7ab378257b8723e02cbceac7410be4db286436"
+    sha1 "ab083063e888527c1a8facdd43e9e4338e63e777"
+  end
+
+  go_resource "github.com/racker/perigee" do
+    url "https://github.com/racker/perigee.git"
```
Comment (DomT4) [inline Formula/docker-machine.rb]: Can you peg these new resources to a specific ` revision ` (i.e. git commit)? It helps prevent dependency updates suddenly causing packages to break, which Go deps (along with Node deps) have a tendency to do. Thanks!

Comment (DanLipsitt) [inline Formula/docker-machine.rb]: Done.

**Comment by DomT4** [comment]
Any chance this is missing a dep? Build is failing locally. (The bot should be fixed soon, so I'll get to stop testing everything manually, but till then...)

```
==> go build -o docker-machine
src/google.golang.org/api/compute/v1/compute-gen.go:17:2: cannot find package "golang.org/x/net/context" in any of:
	/usr/local/Cellar/go/1.4.1/libexec/src/golang.org/x/net/context (from $GOROOT)
	/private/tmp/docker-machine-fKUq6a/machine-0.1.0-rc2/src/golang.org/x/net/context (from $GOPATH)
/usr/local/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: go build -o docker-machine
```

File: Formula/docker-machine.rb (hunk 1 above)
Comment (DomT4) [inline Formula/docker-machine.rb]: Appreciated!

**Comment by DanLipsitt** [comment]
> cannot find package "golang.org/x/net/context"

I did an uninstall and installed again (I'd been using `reinstall`) and it built fine. I also don't see it in Godeps.json. I'm not a go expert, so I'm not sure what's going on. Maybe I've got something in my environment and build isolation isn't working?

**Comment by DanLipsitt** [comment]
I can't find any references to the `net/context` library except in a vendorized copy of the lib itself:

```
$ pushd /Library/Caches/Homebrew/docker-machine--git
/Library/Caches/Homebrew/docker-machine--git /tmp
$ ack net/context
Godeps/_workspace/src/golang.org/x/net/context/context.go
111:	// 	import "golang.org/x/net/context"

Godeps/_workspace/src/golang.org/x/net/context/withtimeout_test.go
11:	"golang.org/x/net/context"
```

**Comment by DomT4** [comment]
Hmm. I'll try a rebuild of ` go ` locally just to check my env isn't out of whack.

**Comment by DomT4** [comment]
Yup, that fixed it. Something about my ` go ` installation was borked by the looks of it. Awesome. Will merge imminently :+1:.

File: Formula/docker-machine.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ class DockerMachine < Formula
   homepage "https://github.com/docker/machine"
 
   devel do
-    url "https://github.com/docker/machine/archive/0.0.2.tar.gz"
-    sha1 "adcc7128abdbaae60ba3c58d6485df939c7e5510"
+    url "https://github.com/docker/machine/archive/v0.1.0-rc2.tar.gz"
+    sha1 "5251d959f6ff1ee6a7e7fb0d7b23aeb6152d7d3c"
```
Comment (DomT4) [inline Formula/docker-machine.rb]: Oh, one last thing. If you could stick a ` version "0.1.0_rc2" ` under the ` sha1 ` that would be awesome. Stops Brew assigning it the wrong version number based on the way Brew handles the ` - ` break between numbers.

**Comment by DanLipsitt** [comment]
I don't think it's the source of your error, but it looks like Godeps.json was [updated today](https://github.com/docker/machine/commit/cdd891ce7c0ef9474c7c60269414af317683bf67). I don't understand how we're supposed to keep go_resources up to speed. Seems like a losing battle. Is it okay to use godep directly since it already freezes dependencies?

File: Formula/docker-machine.rb (hunk 2 above)
Comment (DanLipsitt) [inline Formula/docker-machine.rb]: > under the sha1

I put it under the url so it's visually adjacent to the version number in the tarball.

**Comment by DomT4** [comment]
> Is it okay to use godep directly since it already freezes dependencies?

For better or worse, this is something Homebrew/Homebrew dictates. Taps operate within the upstream framework, implemented since https://github.com/Homebrew/homebrew/pull/29820.

You could always open an issue upstream at Homebrew/Homebrew on this subject, if desired.

**Comment by DomT4** [comment]
Merged, Thanks for the contribution! :smiley_cat:

**Comment by radeksimko** [comment]
> @radeksimko how did you generate the checksums for those resources?

If I remember correctly, I just ran something like `brew audit` or some other commands that test bot runs in verbose mode and then copy-pasted the calculated checksum that did not match.
So to answer your question - I don't know how does it get created for whole repositories (and it seems that it's only required for some of them), but homebrew source code may have the answer.

Generally speaking for go dependencies, there should be an easy way of generating all this, I agree it's pain to update it manually.

Here's what I found so far:
https://groups.google.com/forum/#!topic/golang-nuts/611QEJqkDKw

It might not be hard to write a script that does something like:
```ruby
for path in `deplist github.com/docker/machine`:
if Dir.exists(path+'/.hg'):
  last_revision = `hg #{path} log -l 1`
  url = `hg %get repo url%`
  checksum = ??

  dependency = <<<EOL
go_resource "#{url}" do
    url "https://#{url}",
        :revision => "#{last_revision}"
    sha1 "#{checksum}"
end
EOL
elif Dir.exists(path+'/.git'):
... similar approach for Git
```

**Comment by DanLipsitt** [comment]
@DomT4 it looks like not all of the commits got merged.

**Comment by DomT4** [comment]
@DanLipsitt I squashed them into a single commit prior to merge, to avoid filling up the commit tree with multiple commits for the same formula. Homebrew/Homebrew follows the same merge method. Did I miss something in the commits?

**Comment by DanLipsitt** [comment]
Oh good. I was just going by the message in this PR that says "closed with unmerged commits."

**Comment by DomT4** [comment]
Ah yeah, that's a standard thing with Homebrew. Because we cherry-pick commits rather than merge them directly you'll always see that message, but as long as you see your name in the commit tree here it was merged okay.

Sadly, if we use the "Merge this PR" Github button directly it dumps at least two commits in the commit tree, one attributed to you along the lines of *"docker-machine 0.1.0-rc2"*, and another to me along the lines of *"Merged #11 into master"*, which makes it unnecessarily messy, so we just ` git cherry-pick ` instead which attributes the single commit directly to you without filling up the tree with unnecessary admin commits.

**Comment by DomT4** [comment]
> So to answer your question - I don't know how does it get created for whole repositories (and it seems that it's only required for some of them), but homebrew source code may have the answer.

Any idea on this @mikemcquaid? How are we generating checksums for entire git repositories at certain revision points?

> Generally speaking for go dependencies, there should be an easy way of generating all this, I agree it's pain to update it manually.

The script idea is potentially worth kicking up to Homebrew/Homebrew. I don't know if they consider the way we handle "Go" complete, but the idea of a godeps script seems like a potential improvement to me.

**Comment by mikemcquaid** [comment]
> Any idea on this @mikemcquaid? How are we generating checksums for entire git repositories at certain revision points?

We aren't, the `sha1` is ignored for repositories.

**Comment by DomT4** [comment]
Ha, darn. I was hoping there was some complex and clever code to dive into here :grinning:. Will remove the redundant SHAs shortly.

**Pull request closed without merging**